packages/next/src/server/web/globals.ts

```diff
@@ -0,0 +1,73 @@
+declare const _ENTRIES: any
+
+async function registerInstrumentation() {
+  if (
+    '_ENTRIES' in globalThis &&
+    _ENTRIES.middleware_instrumentation &&
+    _ENTRIES.middleware_instrumentation.register
+  ) {
+    try {
+      await _ENTRIES.middleware_instrumentation.register()
+    } catch (err: any) {
+      err.message = `An error occurred while loading instrumentation hook: ${err.message}`
+      throw err
+    }
+  }
+}
+
+let registerInstrumentationPromise: Promise<void> | null = null
+export function ensureInstrumentationRegistered() {
+  if (!registerInstrumentationPromise) {
+    registerInstrumentationPromise = registerInstrumentation()
+  }
+  return registerInstrumentationPromise
+}
+
+function getUnsupportedModuleErrorMessage(module: string) {
+  // warning: if you change these messages, you must adjust how react-dev-overlay's middleware detects modules not found
+  return `The edge runtime does not support Node.js '${module}' module.
+Learn More: https://nextjs.org/docs/messages/node-module-in-edge-runtime`
+}
+
+function __import_unsupported(moduleName: string) {
```

Comment:
Can you add comments on what this is supposed to do? It looks like your approximating `import()`, but the return value from this is a proxy that return other another proxies for every property access. But, there's no `get` access on `import()`, it's just an `apply`.

Reply:
Oh I should mention that this is just moving files as before they are in enhanceGlobals function, but as moving to ESM, imports are hoisted so we need to import this globally first. Nothing changed but moving the place. I mentioned simply in the description but forget to say there’s no actual code change